code/modules/events/apc_short.dm

```diff
@@ -26,55 +26,95 @@
+	for(var/thing in GLOB.apcs)
+		var/obj/machinery/power/apc/A = thing
+		var/area/current_area = get_area(A)
+		if((current_area.type in skipped_areas_apc) || !is_station_level(A.z))
+			continue
+		var/obj/item/stock_parts/cell/C = A.get_cell()
+		if(C)
+			C.charge = 0
+		current_area.power_change()
+	log_and_message_admins("Power has been drained from all APCs.")
 
+/proc/power_failure(announce=TRUE)
+	// skip any APCs that are too critical to disable
+	var/list/skipped_areas_apc = list(
+		/area/engine/engineering,
+		/area/turret_protected/ai)
 	if(announce)
 		GLOB.event_announcement.Announce("Overload detected in [station_name()]'s powernet. Engineering, please repair shorted APCs.", "Systems Power Failure", new_sound = 'sound/AI/attention.ogg')
-
 	// break APC_BREAK_PROBABILITY% of all of the APCs on the station
 	var/affected_apc_count = 0
 	for(var/thing in GLOB.apcs)
-		var/obj/machinery/power/apc/C = thing
-		// skip any APCs that are too critical to disable
-		var/area/current_area = get_area(C)
-		if((current_area.type in skipped_areas_apc) || !is_station_level(C.z))
+		var/obj/machinery/power/apc/A = thing
+		var/area/current_area = get_area(A)
+		if((current_area.type in skipped_areas_apc) || !is_station_level(A.z))
 			continue
 		// if we are going to break this one
 		if(prob(APC_BREAK_PROBABILITY))
 			// if it has internal wires, cut the power wires
-			if(C.wires)
-				if(!C.wires.is_cut(WIRE_MAIN_POWER1))
-					C.wires.cut(WIRE_MAIN_POWER1)
-				if(!C.wires.is_cut(WIRE_MAIN_POWER2))
-					C.wires.cut(WIRE_MAIN_POWER2)
+			if(A.wires)
+				if(!A.wires.is_cut(WIRE_MAIN_POWER1))
+					A.wires.cut(WIRE_MAIN_POWER1)
+				if(!A.wires.is_cut(WIRE_MAIN_POWER2))
+					A.wires.cut(WIRE_MAIN_POWER2)
 			// if it was operating, toggle off the breaker
-			if(C.operating)
-				C.toggle_breaker()
+			if(A.operating)
+				A.toggle_breaker()
 			// no matter what, ensure the area knows something happened to the power
 			current_area.power_change()
 			affected_apc_count++
-	log_and_message_admins("APC Short event shorted out [affected_apc_count] APCs.")
+	log_and_message_admins("APC Short Out event has shorted out [affected_apc_count] APCs.")
 
-/proc/power_restore(announce=TRUE)
-	if(announce)
-		GLOB.event_announcement.Announce("Power has been restored to [station_name()]. We apologize for the inconvenience.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
-
-	// recharge the APCs
-	for(var/thing in GLOB.apcs)
-		var/obj/machinery/power/apc/A = thing
-		if(!is_station_level(A.z))
-			continue
-		var/obj/item/stock_parts/cell/C = A.get_cell()
-		if(C)
-			C.give(C.maxcharge)
+/proc/power_restore(announce=TRUE, var/power_type)
+	if(power_type == 0)	//Power without Repairing
+		if(announce)
+			GLOB.event_announcement.Announce("All operational APCs on [station_name()] have been fully charged.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
```

Comment:
Changing to `GLOB.event_announcement.Announce("All APCs on \the [station_name()]` for grammar along with other instances.